src/classes/json_data.py

```diff
@@ -305,16 +307,19 @@ def replace_string_to_relative(self, match):
             orig_abs_path = os.path.abspath(path)
 
             # Determine windows drives that the project and file are on
-            project_win_drive = os.path.splitdrive(get_app().project.current_filepath)[0]
+            project_win_drive = os.path.splitdrive(path_context.get("new_project_folder", ""))[0]
```

Comment:
I see... Again safer to use get, and a default value

Reply:
Well, sort of. I mean, yeah, that's kind of always true in Python, unfortunately. It's one of the language's biggest down sides... but in this case, getting an empty or default value wouldn't have been any better. It just wasn't the right place to look for the information we want, that was already supplied in the context dict.

Reply:
(That I'm STILL using `get()` to access that info, when I KNOW it will be set... is an abundance of paranoia.)

Reply:
(There's actually a subclass of `dict`, I think in `collections`, called the `DefaultDict`, that can be configured on creation to return a default value for **any** missing key that's queried. Problem is, it always returns the same default, so to create complex hierarchies you have to instantiate a lot of separate, nested `DefaultDict`s. And even then, it's not very compatible with tree-walking tools like `json.dumps` as it creates lots of phantom values -- if not infinite recursions.)

Reply:
Better than `DefaultDict`, for the project data — and the reason I keep harping on it, mostly to motivate myself to get back to it — would be to have a JSON Schema for the project file (#3556).

A fully realized schema would mean that we could at least _validate_, and potentially repair, project data both out to and in from the file, meaning that damaged project files wouldn't result in damaged `project.data` dicts. It wouldn't protect us from bugs _in the code_ that corrupt the data while it's live, but it would mean we could be certain that any corrupted dicts have to have resulted **from** bugs in the code, hopefully making them easier to track down and fix.